lib/matplotlib/colorbar.py

```diff
@@ -510,166 +490,276 @@ def __init__(self, ax, cmap=None,
+                X, Y, C, cmap=self.cmap, norm=self.norm, alpha=self.alpha,
+                edgecolors='none', shading='flat')
+            if not self.drawedges:
+                if len(self._y) >= self.n_rasterize:
+                    self.solids.set_rasterized(True)
+        self.dividers.set_segments(
+            np.dstack([X, Y])[1:-1] if self.drawedges else [])
+
+    def _add_solids_patches(self, X, Y, C, mappable):
+        hatches = mappable.hatches * len(C)  # Have enough hatches.
+        patches = []
+        for i in range(len(X) - 1):
+            xy = np.array([[X[i, 0], Y[i, 0]],
+                           [X[i, 1], Y[i, 0]],
+                           [X[i + 1, 1], Y[i + 1, 0]],
+                           [X[i + 1, 0], Y[i + 1, 1]]])
+            patch = mpatches.PathPatch(mpath.Path(xy),
+                                       facecolor=self.cmap(self.norm(C[i][0])),
+                                       hatch=hatches[i], linewidth=0,
+                                       antialiased=False, alpha=self.alpha)
+            self.ax.add_patch(patch)
+            patches.append(patch)
+        self.solids_patches = patches
+
+    def _do_extends(self, extendlen):
+        """
+        Make adjustments of the inner axes for the extend triangles (or
+        rectanges) and add them as patches.
+        """
+        # extend lengths are fraction of the *inner* part of colorbar,
+        # not the total colorbar:
+        elower = extendlen[0] if self._extend_lower() else 0
+        eupper = extendlen[1] if self._extend_upper() else 0
+        tot = eupper + elower + 1
+        elower = elower / tot
+        eupper = eupper / tot
+        width = 1 / tot
+
+        bounds = np.array([0.0, elower, 1.0, width])
+
+        # make the inner axes smaller to make room for the extend rectangle
+        top = bounds[1] + bounds[3]
+        if not self.extendrect:
+            # triangle:
+            xyout = np.array([[0, bounds[1]], [0.5, 0], [1, bounds[1]],
+                              [1, top], [0.5, 1], [0, top], [0, bounds[1]]])
+        else:
+            # rectangle:
+            xyout = np.array([[0, bounds[1]], [0, 0], [1, 0], [1, bounds[1]],
+                              [1, top], [1, 1], [0, 1], [0, top],
+                              [0, bounds[1]]])
+        if self.orientation == 'horizontal':
+            bounds = bounds[[1, 0, 3, 2]]
+            xyout = xyout[:, ::-1]
+        self.ax._set_inner_bounds(bounds)
+
+        if not self.filled:
+            return xyout
+        # Make extend triangles or rectangles.  These are defined in the
+        # outer parent axes' co-ordinates:
+        mappable = getattr(self, 'mappable', None)
+        if (isinstance(mappable, contour.ContourSet)
+                and any(hatch is not None for hatch in mappable.hatches)):
+            hatches = mappable.hatches
+        else:
+            hatches = [None]
+        if self._extend_lower:
```

Comment:
This should be a function call.

Reply:
Oh hmmm, interesting I guess those just always ran, but worked fine because `elower` was set to 0...